### PR TITLE
TW-1817  added french and russian translation in camera screen

### PR DESCRIPTION
**Pull request by KhaledNjim**
## Ticket
#1817 

## Root cause
The package used for taking photos (wechat_camera_picker) does not provide translation for russian and french language

## Solution
Added text delegate for russian and french

## Impact description
**If this is not a bug, please explain how the changes affect the project**

## Test recommendations
**Recommendations for how to test this, or anything else you are worried about?**

## Resolved
**Attach screenshots or videos demonstrating the changes**
- Web:
- Android:

[1817.webm](https://github.com/user-attachments/assets/d4369330-cb3d-4656-9e6a-f4fec9afc986)

- IOS:

### CODE REVIEW

File: assets/l10n/intl_en.arb [hunk 1]
```diff
@@ -3086,5 +3086,21 @@
...
+  "shootingOnlyRecordingTips": "Long press to record video.",
+  "shootingTapRecordingTips": "Tap to record video.",
+  "loadFailed": "Load failed",
+  "saving": "Saving...",
+  "sActionManuallyFocusHint": "manually focus",
+  "sActionPreviewHint": "preview",
+  "sActionRecordHint": "record",
+  "sActionShootHint": "take picture",
+  "sActionShootingButtonTooltip": "shooting button",
+  "sActionStopRecordingHint": "stop recording",
```
Comment (nqhhdev) [inline assets/l10n/intl_en.arb:3101]:
```suggestion
  "sActionManuallyFocusHint": "Manually focus",
  "sActionPreviewHint": "Preview",
  "sActionRecordHint": "Record",
  "sActionShootHint": "Take picture",
  "sActionShootingButtonTooltip": "Shooting button",
  "sActionStopRecordingHint": "Stop recording",
```

What do you think?

File: lib/presentation/mixins/common_media_picker_mixin.dart [hunk 2]
```diff
@@ -80,14 +82,36 @@ mixin CommonMediaPickerMixin {
...
+        return LocalizedCameraPickerTextDelegate(
+          context,
+          LocalizationService.currentLocale.value.languageCode,
+        );
+      default:
+        return cameraPickerTextDelegateFromLocale(
+          LocalizationService.currentLocale.value,
+        );
+    }
+  }
```
Comment (nqhhdev) [inline lib/presentation/mixins/common_media_picker_mixin.dart:116]:
```suggestion
  CameraPickerTextDelegate getTextDelegateForLocale(
    BuildContext context,
  ) {
    return LocalizedCameraPickerTextDelegate(
          context,
          LocalizationService.currentLocale.value.languageCode,
        );
  }
```

No need to `switch`, WDYT?

Comment (KhaledNjim) [reply lib/presentation/mixins/common_media_picker_mixin.dart:116]: we get translations from L10n for Russian and French, but for Vietnamese and English, we use the translations provided by the wechat_camera package, which is why we need to switch between them

**Review by nqhhdev — APPROVED** [review]
LGTM

File: lib/presentation/mixins/common_media_picker_mixin.dart [hunk 3]
```diff
@@ -80,14 +82,36 @@ mixin CommonMediaPickerMixin {
...
+    switch (LocalizationService.currentLocale.value.languageCode) {
+      case 'ru':
+      case 'fr':
+        return LocalizedCameraPickerTextDelegate(
+          context,
+          LocalizationService.currentLocale.value.languageCode,
+        );
+      default:
+        return cameraPickerTextDelegateFromLocale(
+          LocalizationService.currentLocale.value,
```
Comment (hoangdat) [inline lib/presentation/mixins/common_media_picker_mixin.dart:113]: it is a huge risk if this locale value is not translated. IMO, fallback case must be `en`

Comment (KhaledNjim) [reply lib/presentation/mixins/common_media_picker_mixin.dart:113]: When the device is set to an unsupported language, `LocalizationService.currentLocale` defaults to English. As a result, camera_picker strings will also be displayed in English

## Demo with device set to the spanish language

[1817_fallback.webm](https://github.com/user-attachments/assets/f1000e73-9241-4691-9640-239d5402e2a8)

**Review by hoangdat — APPROVED** [review]

**Pull request merged**